src/sdk/autodetectcompilers.cpp

```diff
@@ -40,109 +45,240 @@ AutoDetectCompilers::AutoDetectCompilers(wxWindow* parent)
+                    controlLblDefCompiler->SetBackgroundColour(colorDefault);
+                }
+                else
+                {
+                    controlLblDefCompiler->SetLabel(wxString::Format(_("INVALID: %s"),cItem.wxsCompilerName));
+                    controlLblDefCompiler->SetBackgroundColour(*wxRED);
+                }
+            }
+
+            vCompilerList.push_back(cItem);
         }
-        // Resize columns so one can read the whole stuff:
-        list->SetColumnWidth(0, wxLIST_AUTOSIZE);
-        list->SetColumnWidth(1, wxLIST_AUTOSIZE);
+        UpdateCompilerDisplayList();
     }
-
-    XRCCTRL(*this, "lblDefCompiler", wxStaticText)->SetLabel(CompilerFactory::GetDefaultCompiler()->GetName());
 }
 
 AutoDetectCompilers::~AutoDetectCompilers()
 {
     //dtor
 }
 
-void AutoDetectCompilers::OnDefaultClick(cb_unused wxCommandEvent& event)
+void AutoDetectCompilers::OnDefaultCompilerClick(cb_unused wxCommandEvent& event)
 {
     wxListCtrl* list = XRCCTRL(*this, "lcCompilers", wxListCtrl);
-    int idx = list->GetNextItem(-1, wxLIST_NEXT_ALL, wxLIST_STATE_SELECTED);
-    if (idx != -1)
+    int idxList = list->GetNextItem(-1, wxLIST_NEXT_ALL, wxLIST_STATE_SELECTED);
+    if (idxList != -1)
+    {
+        wxString wsSelection = list->GetItemText(idxList);
+        Compiler* compiler = CompilerFactory::GetCompilerByName(wsSelection);
+        int idxComiler = CompilerFactory::GetCompilerIndex(compiler);
+
+        CompilerFactory::SetDefaultCompiler(idxComiler);
+
+        wxColour  colorDefault =  XRCCTRL(*this, "ID_STATICTEXT1", wxStaticText)->GetBackgroundColour();
+        wxStaticText* controlLblDefCompiler = XRCCTRL(*this, "lblDefCompiler", wxStaticText);
+        controlLblDefCompiler->SetLabel(CompilerFactory::GetDefaultCompiler()->GetName());
+        controlLblDefCompiler->SetBackgroundColour(colorDefault);
+    }
+}
+
+void AutoDetectCompilers::OnUdateCompilerListUI(cb_unused wxCommandEvent& event)
+{
+    UpdateCompilerDisplayList();
+}
+
+void AutoDetectCompilers::UpdateCompilerDisplayList()
+{
+    bool bShowCompilerPath= XRCCTRL(*this,"cbShowCompilerPath", wxCheckBox)->GetValue();
+    bool bShowAllCompilerOptions = (XRCCTRL(*this, "rbCompilerShowOptions", wxRadioBox)->GetSelection() == 1);
+
+    wxListCtrl* list = XRCCTRL(*this, "lcCompilers", wxListCtrl);
+    if (list)
     {
-        CompilerFactory::SetDefaultCompiler(idx);
-        XRCCTRL(*this, "lblDefCompiler", wxStaticText)->SetLabel(CompilerFactory::GetDefaultCompiler()->GetName());
+        list->ClearAll();
+        list->InsertColumn(ccnNameColumn, _("Compiler"), wxLIST_FORMAT_LEFT, 380);
+        list->InsertColumn(ccnStatusColumn, _("Status"), wxLIST_FORMAT_LEFT, 100);
+
+        if (bShowCompilerPath)
+            list->InsertColumn(ccnDetectedPathColumn, _("Compiler Path"), wxLIST_FORMAT_LEFT, 200);
+
+        for (auto itCL = std::begin(vCompilerList); itCL != std::end(vCompilerList); ++itCL)
```

Comment:
One more auto...

Reply:
Changed to the following in the next code push:
for (std::vector<CompilerItem>::iterator it = m_CompilerList.begin(); it != m_CompilerList.end(); ++it)

Is this acceptable?

Reply:
The code fix has been pushed.